File: Microsoft/8/8.py
import math

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


def findSecondLastNode(head):
    temp = head

    if (temp == None or temp.next == None):
        return -1

    while (temp != None):

        if (temp.next.next == None):
            return temp.data

        temp = temp.next


def push(head, new_data):
    new_node = Node(new_data)
    # new_node.data = new_data
    new_node.next = head
    head = new_node
    return head


if __name__ == '__main__':
    head = None
    n=int(input())
    x=input()
    inputs = x.split()
    for i in inputs:
        head = push(head, i)

    print(findSecondLastNode(head))

File: Microsoft/8/8.cpp
#include <iostream>
using namespace std;
struct Node {
  int data;
  struct Node *next;
};
int findSecondLastNode(struct Node *head) {
  struct Node *temp = head;
  if (temp == NULL || temp->next == NULL)
    return -1;
  while (temp != NULL) {
    if (temp->next->next == NULL)
      return temp->data;
    temp = temp->next;
  }
}

void push(struct Node **head_ref, int new_data) {
  Node *new_node = new Node;
  new_node->data = new_data;
  new_node->next = (*head_ref);
  (*head_ref) = new_node;
}

int main() {
  struct Node *head = NULL;
  int n;
  cin >> n;
  for (int i = 0; i < n; i++) {
    int j;
    cin >> j;
    push(&head, j);
  }

  cout << findSecondLastNode(head);

  return 0;
}
